pkg/controller/tenant/tenant_reconcile.go

```diff
@@ -42,6 +47,14 @@ func (r *ReconcileTenant) Reconcile(request reconcile.Request) (reconcile.Result
 		instance.Spec.GitRepoTemplate.DisplayName = instance.Spec.DisplayName
 	}
 
+	commonClassFile := CommonClassName + ".yml"
+	if instance.Spec.GitRepoTemplate.TemplateFiles == nil {
```

Comment:
This could lead to a race condition with the logic in the `ReconcileCluster()` function, as they both manipulate this map and want to write it back to the K8s API. Could lead to those dreadful `the object has been modified` errors.

We may have to wrap the whole reconcile into a `retry` function to avoid that.